dig.go

```diff
@@ -170,6 +176,78 @@ func (c *Container) provide(ctor interface{}, ctype reflect.Type) error {
 	return nil
 }
 
+func (c *Container) findConnections(n *node) (map[key]struct{}, error) {
+	var err error
+	keyPaths := make(map[key]string)
+	walkResult(n.Results, connectionVisitor{
+		c:    c,
+		n:    n,
+		err:  &err,
+		keys: keyPaths,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[key]struct{}, len(keyPaths))
```

Comment:
we throw this away immediately,  should we just return the keyPaths?

Reply:
They have different purposes. This is the set that we're expecting in the caller and `keyPaths` is used for tracking by the connectionVisitor. I'll document the individual fields of connectionVisitor further.